pkg/operator/k8sutil/node.go

```diff
@@ -246,3 +247,41 @@ func NodeIsInRookNodeList(targetNodeName string, rookNodes []rookalpha.Node) boo
 	}
 	return false
 }
+
+// AddNodeAffinity will return a v1.NodeAffinity object
+func AddNodeAffinity(nodeAffinity string) *v1.NodeAffinity {
+	newNodeAffinity := &v1.NodeAffinity{
+		RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+			NodeSelectorTerms: []v1.NodeSelectorTerm{
+				{},
+			},
+		},
+	}
+	// Replace all whitespaces in the string 'nodeAffinity'
+	// Retrieve Labels from the string
+	// ';' separates labels
+	nodeAffinity = strings.Replace(nodeAffinity, " ", "", -1)
+	nodeAffinity = strings.Replace(nodeAffinity, ";", " ", -1)
+	nodeLabels := strings.Fields(nodeAffinity)
+	// For each label in 'nodeLabels', retrieve (key,value) pair and create nodeAffinity
+	// '=' separates key from values
+	// ',' separates values
+	for _, nodeLabel := range nodeLabels {
+		// tmpNodeLabel will always be an array of length 2
+		// [0] is Key and [1] is comma separated values
+		tmpNodeLabel := strings.Fields(strings.Replace(nodeLabel, "=", " ", 1))
+		nodeLabelKey := tmpNodeLabel[0]
```

Comment:
If the nodeLabel doesn't contain the `=`, you will get an out of range exception for `tmpNodeLabel[1]`

Reply:
Yes. Currently, the assumption is that user will always provide a `key=value` pair. Checking for presence of `key` is something I tried but it makes the parsing of whole ENV Var a little complicated.
I will look at the exception handling.